src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -166,12 +166,33 @@ public override AppendPrependIterator<TSource> Prepend(TSource item)
                 }
             }
 
+            private TSource[] LazyToArray()
+            {
+                Debug.Assert(GetCount(onlyIfCheap: true) == -1);
+
+                var builder = new LargeArrayBuilder<TSource>(initialize: true);
+                
+                if (!_appending)
+                {
+                    builder.SlowAdd(_item);
+                }
+
+                builder.AddRange(_source);
+
+                if (_appending)
+                {
+                    builder.SlowAdd(_item);
+                }
```

Comment:
@stephentoub, this is the same pattern that is used in other places throughout the file, e.g. [here](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/AppendPrepend.cs#L211).